audio-worklet/test/bitcrusher-test.js

```diff
@@ -35,79 +35,95 @@ for (let factor = 1; factor < 10; factor++) {
-bitcrusher = new Bitcrusher(offlineContext, {
-  buffersize: 512,
+let merger = new ChannelMergerNode(context, {numberOfInputs: 2});
+bitcrusher = new Bitcrusher(context, {
+  buffersize: bufferLength,
   inputChannels: 1,
-  outputChannels: 1, bitDepth, reduction
+  outputChannels: 1, 
+  bitDepth: bitDepth, 
+  reduction: reduction
 });
 
-osc.connect(bitcrusher.input);
+// Accomodate for script processor latency by delaying the unaltered oscillator.
+let delay = context.createDelay();
+delay.delayTime.value = bufferLength / context.sampleRate; 
+
+bitcrusherOscillator.connect(bitcrusher.input);
 bitcrusher.output.connect(merger, 0, 0);
-oscb.connect(merger, 0, 1);
-merger.connect(offlineContext.destination);
+unalteredOscillator.connect(delay).connect(merger, 0, 1);
+merger.connect(context.destination);
 
 // When audio buffer is ready, verify bitcrushed samples are unaltered.
-offlineContext.startRendering()
-    .then(function(buffer) {
-      let bcOutput = buffer.getChannelData(0);
-      let oscOutput = buffer.getChannelData(1);
+context.startRendering()
+    .then((buffer) => {
+      let bitcrusherOutput = buffer.getChannelData(0);
+      let unalteredOutput = buffer.getChannelData(1);
 
-      // Allow for fractional error beyond audible perception.
-      const permittedSampleError = 0.0000001;
+      // Allow for fractional error beyond audible perception. This error
+      // occurs because any sample passing through bitcrusher will undergo 
+      // manipulation in Math.round() and will therefore be represented by a
+      // new floating point number that differs slightly from the original. 
+      // In tested  samples, the maximum observed error is on the order of 10^-7 
+      // and most frequently between 0 and 10^-8. The permitted error is set
+      // here to be 10^-6 so that tests pass.
+      const permittedSampleError = Math.pow(10,-6);
 
       // First script-processed-buffer is all zero.
       const latency = bufferLength;
 
       // Verify samples from unadultered oscillator match samples from
       // bitcrushed oscillator with non-information reducing parameters.
-      for (let i = 0; i < oscOutput.length - latency; i++) {
-        let crushedSample = bcOutput[i + latency];
-        let oscSample = oscOutput[i];
-        const diff = Math.abs(oscSample - crushedSample);
+      for (let i = 0; i < unalteredOutput.length; i++) {
+        let crushedSample = bitcrusherOutput[i];
+        let unalteredSample = unalteredOutput[i];
+        const diff = Math.abs(unalteredSample - crushedSample);
         console.assert(
-            permittedSampleError > diff,
+            diff < permittedSampleError,
             'Bitcrushed sample at ' + i + ' is ' + crushedSample + ' but ' +
-                oscSample + ' for oscillator');
+                unalteredSample + ' for oscillator');
       }
     })
-    .catch(function(err) {
+    .catch((err) => {
       console.log('Rendering failed: ' + err);
```

Comment:
`startRendering` can't really fail in any way, so this is pretty useless, I think.